File: scripts/inspect-login-page.py
import requests
url = 'https://practicetestautomation.com/practice-test-login/'
r = requests.get(url, timeout=30)
print(r.status_code)
html = r.text
print('username=', 'id="username"' in html)
print('password=', 'id="password"' in html)
print('submit_button=', 'type="submit"' in html)
print('button_text=', 'Submit' in html)
print('form=', '<form' in html)
idx = html.find('id="username"')
if idx != -1:
    print(html[idx-120:idx+200])
else:
    print('username element not found')

File: scripts/inspect-login-page.js
const https = require('https');
const url = 'https://practicetestautomation.com/practice-test-login/';
https.get(url, (res) => {
  let data = '';
  res.on('data', (c) => { data += c; });
  res.on('end', () => {
    console.log(res.statusCode);
    console.log('username=', data.includes('id="username"'));
    console.log('password=', data.includes('id="password"'));
    console.log('submit_button=', data.includes('type="submit"'));
    console.log('button_text=', data.includes('Submit'));
    console.log('form=', data.includes('<form'));
    const idx = data.indexOf('id="username"');
    console.log(idx !== -1 ? data.slice(Math.max(0, idx - 120), idx + 200) : 'username not found');
  });
}).on('error', (e) => {
  console.error(e);
  process.exit(1);
});
